src/js/angular/core/directives/multiselect-dropdown/multiselect-dropdown.directive.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @function multiselectDropdown
+ * @description
+ * A directive for rendering a multi-select dropdown.
+ *
+ * This directive displays a list of options with checkboxes that allows users
+ * to select multiple items. Clicking the toggle button opens/closes the dropdown;
+ * clicking outside the dropdown closes it.
+ *
+ * @example
+ * ### HTML Usage:
+ * ```html
+ * <multiselect-dropdown
+ *   options="myOptions"
+ *   selected-options="mySelectedOptions"
+ *   dropdown-label="'Choose Items'">
+ * </multiselect-dropdown>
+ * ```
+ *
+ * ### Controller Example:
+ * ```javascript
+ * angular.module('myApp', [])
+ *   .controller('ExampleController', function($scope) {
+ *     $scope.myOptions = ['Option A', 'Option B', 'Option C'];
```

Comment:
So, the options model is basically a string array if I understand it correctly? I'd preffer if we have a model which is a bit more complex than this. For example. For the single select menu we use this model https://github.com/Ontotext-AD/graphdb-workbench/blob/master/src/js/angular/models/form-fields.js#L4 which gives us more flexibility and declarative approch on controlling the menu behavior. For example, I can have options selected by just iterating the model and binding to the `selected` property in the  model instead of relying on a function to resolve it.

Reply:
done